modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/Treeview.js

```diff
@@ -84,7 +84,12 @@ function filterNodes(nodes, filterQuery) {
 	const filteredNodes = [];
 
 	nodes.forEach((node) => {
-		if (node.name.toLowerCase().indexOf(filterQuery) !== -1) {
+		const isSearchable = node.searchable ?? true;
```

Comment:
I don't have a better idea right away and probably lack a lot of context but this feels a bit of an ad-hoc API that will fail to deliver soon enough...

Is it guaranteed that you'll always be able to add the `searchable` attribute to every node in your model? This might be coming from a remote request which we might have no control over.

Not sure if possible, but could we make the `filterFn` the prop instead? I see you already have something like `vocabulary: true`, which maybe we could use together with a custom filter function to avoid adding those to the search?

Reply:
I guess in those cases you could decorate the response 🤔  but nonetheless, I think that your suggestions would make it more flexible, so I'll go for it :) Thanks!